packages/opentelemetry-instrumentation-llamaindex/tests/test_agents.py

```diff
@@ -23,17 +24,57 @@ def multiply(a: int, b: int) -> int:
     spans = exporter.get_finished_spans()
 
     assert {
-        "ReActAgent.agent",
-        "FunctionTool.tool",
+        "AgentRunner.llama_index.workflow",
+        "AgentRunner.llama_index.task",
+        "FunctionTool.llama_index.task",
+        "OpenAI.llama_index.task",
+        "ReActOutputParser.llama_index.task",
+        "ReActAgentWorker.llama_index.task",
         "openai.chat",
-    } == set([span.name for span in spans])
-
-    agent_span = next(span for span in spans if span.name == "ReActAgent.agent")
-    function_tool = next(span for span in spans if span.name == "FunctionTool.tool")
-    openai_span = next(span for span in spans if span.name == "openai.chat")
+    } == {span.name for span in spans}
+
+    agent_workflow_span = next(span for span in spans if span.name == "AgentRunner.llama_index.workflow")
+    function_tool_span = next(span for span in spans if span.name == "FunctionTool.llama_index.task")
+    llm_span_1, llm_span_2 = [span for span in spans if span.name == "OpenAI.llama_index.task"]
+    openai_span_1, openai_span_2 = [span for span in spans if span.name == "openai.chat"]
+
+    assert agent_workflow_span.parent is None
+    assert function_tool_span.parent is not None
```

Comment:
can we assert on the specific structure here? for example that the function tool span's parent is the agent workflow? (or whatever it actually is)

Reply:
Yes, I wanted to do this. But the test would become quite brittle. Below is the span structure for this test. It's very verbose, lot of same named spans. E.g. FunctionTool, the one you mentioned, has ReActAgentWorker as parent, but that pops up 4 times. I am still undecided what is the best approach here - maybe there is a third better option? So please feel free to change it to your liking.

name span_id parent.span_id
ReActAgentWorker.llama_index.task 319464811280482099 7923477847642805764
AgentRunner.llama_index.task 7923477847642805764 5348156688623149641
openai.chat 10833227178492975942 787045099616181983
OpenAI.llama_index.task 787045099616181983 13446663969238565076
ReActOutputParser.llama_index.task 1599754872828344697 13446663969238565076
FunctionTool.llama_index.task 10958318651987115747 13446663969238565076
ReActAgentWorker.llama_index.task 13446663969238565076 15738578304360324053
AgentRunner.llama_index.task 15738578304360324053 5348156688623149641
openai.chat 12351434778324108193 14394044961538896963
OpenAI.llama_index.task 14394044961538896963 18062227453713671516
ReActOutputParser.llama_index.task 17662053697512896373 18062227453713671516
ReActAgentWorker.llama_index.task 18062227453713671516 8217273878331094304
AgentRunner.llama_index.task 8217273878331094304 5348156688623149641
ReActAgentWorker.llama_index.task 7685564445797887478 11969873770185106285
AgentRunner.llama_index.task 11969873770185106285 5348156688623149641
AgentRunner.llama_index.task 5348156688623149641 10450133951577012365
AgentRunner.llama_index.workflow 10450133951577012365 None